src/CodeGen_X86.cpp

```diff
@@ -132,6 +132,12 @@ const x86Intrinsic intrinsic_defs[] = {
     {"llvm.x86.avx512.pmaddw.d.512", Int(32, 16), "pmaddwd", {Int(16, 32), Int(16, 32)}, Target::AVX512_Cannonlake},
     {"llvm.x86.avx2.pmadd.wd", Int(32, 8), "pmaddwd", {Int(16, 16), Int(16, 16)}, Target::AVX2},
     {"llvm.x86.sse2.pmadd.wd", Int(32, 4), "pmaddwd", {Int(16, 8), Int(16, 8)}},
+
+    // Convert FP32 to BF16
+    {"llvm.x86.avx512bf16.cvtneps2bf16.512", BFloat(16, 16), "f32_to_bf16", {Float(32, 16)}, Target::AVX512_SapphireRapids},
+    {"llvm.x86.avx512bf16.cvtneps2bf16.256", BFloat(16, 8), "f32_to_bf16", {Float(32, 8)}, Target::AVX512_SapphireRapids},
+    // LLVM 12 does not support cvtneps2bf16 for 128bit inputs
```

Comment:
LLVM 12 just forked yesterday (LLVM main is now v13); do these changes apply to both

Reply:
The intrinsic is not available in either version. The masked intrinsic is available, so we can provide a wrapper that uses that, similar to some of the other intrinsics here.